client/js/lounge.js

```diff
@@ -641,38 +641,13 @@ $(function() {
 	});
 
 	chat.on("msg", ".messages", function(e, target, msg) {
-		var button = sidebar.find(".chan[data-target=" + target + "]");
-		var isQuery = button.hasClass("query");
-		var type = msg.type;
-		var highlight = type.contains("highlight");
-		var message = type.contains("message");
-		var settings = $.cookie("settings") || {};
-		if (highlight || isQuery || (settings.notifyAllMessages && message)) {
-			if (!document.hasFocus() || !$(target).hasClass("active")) {
-				if (settings.notification) {
-					pop.play();
-				}
-				favico.badge("!");
-				if (settings.badge && Notification.permission === "granted") {
-					var notify = new Notification(msg.from + " says:", {
-						body: msg.text.trim(),
-						icon: "/img/logo-64.png",
-						tag: target
-					});
-					notify.onclick = function() {
-						window.focus();
-						button.click();
-						this.close();
-					};
-					window.setTimeout(function() {
-						notify.close();
-					}, 5 * 1000);
-				}
-			}
+		if (msg.self) {
+			return;
 		}
 
-		button = button.filter(":not(.active)");
-		if (button.length === 0) {
+		var button = sidebar.find(".chan[data-target=" + target + "]");
+		var isActive = button.hasClass("active");
+		if (isActive && document.hasFocus()) {
```

Comment:
I just tried `hasFocus`, and it would be quite annoying actually. Maybe `document.visibilityState` should be used instead?

https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API